src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -292,12 +283,35 @@ private bool ValidateDocument(LazyInput lazyInput, InputValidation inputValidati
             }
         }
 
+        private InputTrace CheckIfTraceIsConfigured(Flow flow, Identity user, Document input)
+        {
+            var message = EnvelopeReceiverContext<Message>.Envelope;
+
+            if (message?.Metadata != null &&
+                message.Metadata.Keys.Contains("builder.trace.target"))
+            {
+                flow.TraceSettings = new TraceSettings(message.Metadata);
```

Comment:
Do not change the `flow .TraceSettings` property because the object is singleton and all subsequent call will have this trace configuration. Create a local variable for that.